code/modules/materials/_materials.dm

```diff
@@ -834,4 +838,9 @@ INITIALIZE_IMMEDIATE(/obj/effect/gas_overlay)
 			.[interaction] = interacted_units
 			total_interacted_units -= interacted_units
 		if(total_interacted_units <= 0)
-			return
\ No newline at end of file
+			return
+
+/decl/material/proc/add_burn_product(var/datum/gas_mixture/environment, var/amount)
```

Comment:
I feel like `add_burn_product` should take `loc` instead of `environment`, so that (as mentioned by the TODO comment about it) it can handle solid products as well?